lib/let_me/policy.ex

```diff
@@ -376,8 +376,11 @@ defmodule LetMe.Policy do
 
   The error reason can be customized by setting the `:error_reason` option when
   using the module.
+
+  The last parameter is a set of arguments that can be defined dynamically
+  which will be passed into any `pre_hook`s defined on the resource's policy.
   """
-  @callback authorize(atom, any, any) :: :ok | {:error, any}
+  @callback authorize(atom, any, any, any) :: :ok | {:error, any}
```

Comment:
With your changes of the `build_pre_hook_calls` macro, the fourth argument is expected to be an `Enumerable.t()`. But I also think we can save us the `Enum.into` and enforce a `keyword`.